rep-2011.rst

```diff
@@ -321,7 +321,214 @@ The passed ``rosidl_message_type_support_t`` in the init call can be introspecte
+
+    rcl_type_introspection_t * introspection_handle;
+    introspection_handle->init();  // Locate local plugins here
+
+    // Plugin name: "fastcdr_v1_0_24"
+    const char * plugin_name = introspection_handle->match_plugin(FooDescription->get_serialization_format());
+    rcl_serialization_plugin_t * plugin = introspection_handle->load_plugin("fastcdr_v1_0_24");
+
+    // If we wanted to force the use of MicroCDR instead
+    introspection_handle->match_plugin(FooDescription->get_serialization_type(), "microcdr");
+
+Then, the plugin should be able to use the description to deserialize the message buffer using the plugin:
+
+.. code::
+
+    rcl_deserialized_message_t * foo_msg;
+    introspection_handle->deserialize(&plugin, message_buffer, &FooDescription, foo_msg);
+
+    // Where, internally...
+    // rcl_type_introspection_t->(*deserialize)
+    void deserialize(rcl_serialization_plugin_t *plugin,
+                     void *buffer,
+                     rcl_message_description_t *description,
+                     rcl_deserialized_message_t *msg)
+    {
+      ...
+
+      // Do something serialization specific
+      plugin_internal_description_t parsed_description = plugin->impl->parse_description(description);
+      plugin->impl->deserialize(buffer, parsed_description, msg);
+
+      ...
+    }
+
+Plugin Deserialization
+""""""""""""""""""""""
+
+One way a serialization library plugin for FastCDR could be implemented could be as follows, relying on the serialization library to deserialize the message buffer, but with the provided type description to supply type reflection information:
+
+.. code::
+
+    #include <FastBuffer.h>
+    #include <FastCdr.h>
+
+    // If the parsed description is iterable
+    // rcl_serialization_plugin_t->(*deserialize)
+    void deserialize(void *buffer,
+                     plugin_internal_description_t *description,
+                     rcl_deserialized_message_t *msg)
+    {
+      FastCdr * cdr = new FastCdr(new FastBuffer((char *)buffer, description->buffer_length));
+
+      msg->message_field_names = new char*[description->num_fields];
+      msg->message_types = new char*[description->num_fields];
+
+      for (int i = 0; i < description->num_fields; i++)
+      {
+        const char * type = description->field_types[i];
+
+        // Primitives
+        if (strcmp(type, "bool") == 0)
+        {
+          bool tmp;
+        }
+        else if (strcmp(type, "char") == 0)
+        {
+          char tmp;
+        }
+        ...
+
+        // TODO:: What to do about nested types? A recursive unpacking call with a different description?
+
+        cdr >> tmp;
+        msg->message_fields[description->field_names[i]] = reinterpret_cast<void *>(tmp);
+
+        msg->message_field_names[i] = description->field_names[i];
+        msg->message_types[i] = description->field_types[i];
+      }
+      msg->message_field_count = description->num_fields;
+    }
```

Comment:
Based on off-line discussion, let's drop this section.

Reply:
https://github.com/wjwwood/rep/pull/3/commits/bdfa2cfcd926334b1f28845d91b66867ab9218a6